src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1486,22 +1486,28 @@ public int Compare(Symbol fst, Symbol snd)
                         var srcSymbol = symbols[best.Index];
                         var mdSymbol = symbols[secondBest.Index];
 
-                        object arg0;
-
-                        if (best.IsFromSourceModule)
-                        {
-                            arg0 = srcSymbol.Locations.First().SourceTree.FilePath;
-                        }
-                        else
-                        {
-                            Debug.Assert(best.IsFromAddedModule);
-                            arg0 = srcSymbol.ContainingModule;
-                        }
-
                         //if names match, arities match, and containing symbols match (recursively), ...
                         if (srcSymbol.ToDisplayString(SymbolDisplayFormat.QualifiedNameArityFormat) ==
                             mdSymbol.ToDisplayString(SymbolDisplayFormat.QualifiedNameArityFormat))
                         {
+                            if (srcSymbol.Equals(Compilation.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_NonNullTypesAttribute)))
+                            {
+                                // Silently prefer the injected symbol
+                                return originalSymbols[best.Index];
```

Comment:
>return originalSymbols[best.Index]; [](start = 32, length = 35)

What happens if ```secondBest.IsFromCompilation```, but from different one?

Reply:
`IsFromCompilation` is `return (_location == BestSymbolLocation.FromSourceModule) || (_location == BestSymbolLocation.FromAddedModule);`, so it would not be from a different compilation, I think.

---
In reply to: [215092219](https://github.com/dotnet/roslyn/pull/29180#discussion_r215092219) [](ancestors = 215092219)